Hosted CE doc (SOFTWARE-3445)

edquist opened the pull request:
Based on @matyasselmeci's #476.

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,135 @@
+This will allow a broader set of jobs to run on your cluster.
+
+
+Technical Setup
```
```suggestion
Providing the OSG Worker Node client (HTCondor batch systems only)
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,135 @@
+Technical Setup
+---------------
+
+An OSG Hosted CE site needs to have access to the worker node client.
```
All OSG sites need to provide the worker node client

edquist replied:
(updated)

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,135 @@
+users home directories, if this is on a shared filesystem.
+But for sites with an HTCondor batch system, often there is no shared filesystem set up.
+In that case, the site can either set up a shared filesystem, or install the OSG worker node client.
+This can be obtained from the osg-wn-client rpm, the osg-wn tarball, or used directly from OASIS.
```
Even though it's the next section, turn OASIS into a link to it

brianhlin replied:
We should also say something along the lines of: if it is easier for your site to install and maintain the WN client than to set up a shared file system, you have the following options:

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,135 @@
+
+An OSG Hosted CE site needs to have access to the worker node client.
+On a Hosted CE with a non-HTCondor batch system, this is often done by unpacking the worker node client tarball into the
+users home directories, if this is on a shared filesystem.
```
I think we want to hide some of the gory details from the user and say something along the lines that this is normally handled by OSG staff in the hosted CE case but it requires home directories shared across the cluster.

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,138 @@
+[install the OSG worker node client](/worker-node/install-wn).
+This can be obtained from the [osg-wn-client rpm](/worker-node/install-wn),
+the [osg-wn tarball](/worker-node/install-wn-tarball),
+or installed directly from [OASIS](/worker-node/install-wn-oasis).
```
I think I'd prefer this in a bulleted list

edquist commented:
@brianhlin, I think I've addressed the comments you've made in this PR.

In Mat's #476, you also mention another step at the end:
>Let the OSG know that the site is ready, ramp up job sub numbers (probably doesn't need its own subsection)

I have not included this step, and could use some help/discussion about what all we want to say there, if we want to add this section too.

Also note that since I've based the current PR off of Mat's PR head commit, merging this PR will also automatically merge his.

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+However, for sites with an HTCondor batch system, often there is no shared filesystem set up.
+If it is easier for your site to install and maintain the worker node client than to set up a shared file system, you have the following options:
+
+-   Install the [osg-wn-client rpm](/worker-node/install-wn)
```
```suggestion
-   Install the [Worker Node Client from RPM](/worker-node/install-wn)
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+
+-   Install the [osg-wn-client rpm](/worker-node/install-wn)
+
+-   Install the [osg-wn tarball](/worker-node/install-wn-tarball)
```
```suggestion
-   Install the [Worker Node Client from tarball](/worker-node/install-wn-tarball)
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+
+-   Install the [osg-wn tarball](/worker-node/install-wn-tarball)
+
+-   Install the worker node client from [OASIS](/worker-node/install-wn-oasis)
```
```suggestion
-   Install the Worker Node Client from [OASIS](/worker-node/install-wn-oasis)
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+Providing the OSG Worker Node client (HTCondor batch systems only)
+------------------------------------------------------------------
+
+All OSG sites need to provide the worker node client.
```
```suggestion
All OSG sites need to provide the OSG Worker Node Client on each worker node in the cluster.
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+All OSG sites need to provide the worker node client.
+This is normally handled by OSG staff for a Hosted CE, but requires shared home directories across the cluster.
+However, for sites with an HTCondor batch system, often there is no shared filesystem set up.
+If it is easier for your site to install and maintain the worker node client than to set up a shared file system, you have the following options:
```
```suggestion
If it is easier for your site to install and maintain the Worker Node Client on each worker node than to set up a shared file system, you have the following options:
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+
+Before making any system changes, you should do the following steps:
+
+-   Fill out the [cluster integration questionnaire](http://goo.gl/forms/8OukxsyG6KBSGHuR2)
```
Updated link (unfortunately the short-form seems to expire after some time)

```suggestion
-   Fill out the [cluster integration questionnaire](https://docs.google.com/forms/d/e/1FAIpQLSf_pJ3Xlf2N9v5I1GK9oA9HVIxNeRikEq_T1GY0ijaUMy9cIA/viewform)
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+    so it must also have permissions to submit jobs to the batch system
+
+-   If your batch system is not HTCondor,
+    there must be a shared file system between the submit server and the compute nodes
```
Let's add a `See this section for details` with a link to the section you added below

edquist replied:
updated

edquist replied:
updated

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+wherein the OSG team will host and operate the HTCondor compute element,
+and configure it for the science communities that you choose to support.
+
+This document explains the requirements and the procuedure for obtaining a hosted CE.
```
Could you run a spell-checker?

```suggestion
This document explains the requirements and the procedure for obtaining a hosted CE.
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+As an alternative, OSG offers a hosted CE option
+wherein the OSG team will host and operate the HTCondor compute element,
+and configure it for the science communities that you choose to support.
+
```
Let's add a sentence here saying that the Hosted-CE can support thousands of concurrent job submissions

edquist replied:
updated

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+
+This document explains the requirements and the procuedure for obtaining a hosted CE.
+If you wish to have a local compute element instead,
+see page for [installing HTCondor-CE](/compute-element/install-htcondor-ce).
```
Let's change this to something like:

> If you wish to run your own local compute element or expect to support more than 10,000 concurrently running OSG jobs, see this page for...

edquist replied:
updated

edquist replied:
updated

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+
+Optionally, we can assist you in installing and setting up the Squid and OASIS software on your cluster
+in order to support application software repositories.
+This will allow a broader set of jobs to run on your cluster.
```
Link to the section below about OASIS somewhere in this paragraph

edquist replied:
updated

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+
+-   Create a Unix account on the submit server for use by the OSG CE
+
+-   Install public SSH keys for the account
```
```suggestion
-   Install OSG-provided public SSH keys for the account
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+
+After the consultation, do the following:
+
+-   Create a Unix account on the submit server for use by the OSG CE
```
```suggestion
-   Create a Unix account for OSG jobs on the submit server and across the cluster as necessary
```

brianhlin commented on docs/compute-element/hosted-ce.md:
```diff
@@ -0,0 +1,139 @@
+
+Once this is done, the OSG team will:
+
+-   Configure the OSG CE with your system details
```
Let's replace OSG CE with "Hosted CE" everywhere in the doc and make sure Hosted CE is capitalized consistently

edquist replied:
Updated

brianhlin commented:
@edquist there are just a few minimal changes left, let's get 'er done

brianhlin commented:
LGTM!
